server/src/api/feature/__tests__/feature.test.ts

```diff
@@ -1,27 +1,135 @@
-import mysql from 'mysql2';
+import mysql from 'mysql2'
 
-import Config from '../../../config';
-import FeatureController from '../controller';
-import FeatureDAO from '../model';
+import Config from '../../../config'
+import FeatureController from '../controller'
+import FeatureDAO from '../model'
 
-const config = Config.readFromEnvironment();
-const db = mysql.createConnection(config.dbOptions());
-const controller = new FeatureController(new FeatureDAO(db));
+const config = Config.readFromEnvironment()
+const db = mysql.createConnection(config.dbOptions())
+const controller = new FeatureController(new FeatureDAO(db))
 
 test('create feature works as intended', async () => {
   const featureId = await controller.createFeature({
     name: 'test',
     active: false,
-  });
+  })
 
-  const features = await controller.listFeatures();
-  expect(features).toEqual([{
+  const features = await controller.listFeatures()
+  expect(features).toEqual([
+    {
+      id: featureId,
+      name: 'test',
+      active: 0, // booleans get converted to integers
+    },
+  ])
+})
+
+test('update feature works as intended', async () => {
+  const featureId = await controller.createFeature({
+    name: 'test-update',
+    active: false,
+  })
+
+  await controller.updateFeature({
     id: featureId,
-    name: 'test',
-    active: 0, // booleans get converted to integers
-  }]);
-});
+    active: true,
+  })
+
+  const feature = await controller.getFeature(featureId)
+  expect(feature).toEqual({
+    id: featureId,
+    name: 'test-update',
+    deleted_at: null,
+    active: 1,
+  })
+})
+
+test('delete feature works as intended', async () => {
+  const featureId = await controller.createFeature({
+    name: 'test-delete',
+    active: false,
+  })
+
+  await controller.deleteFeature({
+    id: featureId,
+  })
+
+  const feature = await controller.getFeature(featureId)
+  expect(feature.deleted_at).toBeDefined()
+})
+
+test('list features works as intended', async () => {
+  const featureId = await controller.createFeature({
+    name: 'test-list',
+    active: false,
+  })
+
+  await controller.deleteFeature({
+    id: featureId,
+  })
+
+  const feature = await controller.listFeatures()
+  expect(feature).toEqual([
+    {
+      id: 1,
+      name: 'test',
+      active: 0,
+    },
+    {
+      id: 2,
+      name: 'test-update',
+      active: 1,
+    },
+  ])
+})
+
+test('list features combined with delete works as intended', async () => {
+  const featureId = await controller.createFeature({
+    name: 'combo-breaker-list',
+    active: false,
+  })
+
+  await controller.deleteFeature({
+    id: featureId,
+  })
+
+  await controller.deleteFeature({
```

Comment:
can remove this second delete call

Reply:
It seems to fail if I remove that since `combo-breaker-list` has the id of 5. So would I need to keep it in order to delete the the feature 1 and leave only feature 2? 
I probably did not write it in the most straightforward way...

Reply:
What if you just check for individual features in the list (comment as below)